torch/csrc/jit/runtime/register_prim_ops.cpp

```diff
@@ -1494,524 +1494,660 @@ int64_t stringFindImpl(
+          auto pos = string.find(separator, 0);
+          auto rpos = pos;
+          do {
+            pos = rpos;
+            rpos = string.find(separator, pos + 1);
+          } while (rpos != std::string::npos);
+
+          if (pos == std::string::npos) {
+            pos = 0;
+            separator = "";
+          }
+
+          auto pre_partition = string.substr(0, pos);
+          auto post_partition =
+              string.substr(pos + separator.size(), string.size());
+          push(stack, pre_partition, separator, post_partition);
+        },
+        aliasAnalysisFromSchema()),
+    OperatorGeneratorArgs(
+        TORCH_SELECTIVE_SCHEMA(
+            "aten::rsplit(str self, str separator=' ', int max=-1) -> str[]"),
+        [](Stack& stack) {
+          int64_t max = pop(stack).toInt();
+          std::string separator = pop(stack).toStringRef();
+          std::string string = pop(stack).toStringRef();
+          std::reverse(separator.begin(), separator.end());
+          std::reverse(string.begin(), string.end());
+
+          std::string::size_type prev_pos = 0;
+          std::string::size_type pos = 0;
+          c10::List<std::string> splits;
+          auto count = 0;
+          while ((pos = string.find(separator, pos)) != std::string::npos) {
+            count++;
+            if (max >= 0 && count > max) {
+              break;
+            } else {
+              auto substr = string.substr(prev_pos, pos - prev_pos);
+              std::reverse(substr.begin(), substr.end());
+              splits.emplace(splits.begin(), substr);
+            }
+            pos += separator.size();
+            prev_pos = pos;
+          }
+          auto substr = string.substr(prev_pos, string.size() - prev_pos);
+          std::reverse(substr.begin(), substr.end());
+          splits.emplace(splits.begin(), substr);
+          push(stack, std::move(splits));
+        },
+        aliasAnalysisFromSchema()),
+    OperatorGeneratorArgs(
+        TORCH_SELECTIVE_SCHEMA("aten::join(str self, str[] values) -> str"),
+        [](Stack& stack) {
+          IValue ivalue = pop(stack);
+          c10::ArrayRef<IValue> ivalues = ivalue.toListRef();
+          c10::List<std::string> values;
+          for (const auto& v : ivalues) {
+            values.emplace_back(v.toStringRef());
+          }
+          c10::optional<std::string> opt_string =
+              pop(stack).toOptional<std::string>();
+          const std::string& string = opt_string.value_or("");
+          std::stringstream ss;
+          for (auto it = values.begin(); it != values.end(); ++it) {
+            ss << static_cast<std::string>(*it);
+            if (it != values.end() - 1) {
+              ss << string;
+            }
+          }
+          push(stack, ss.str());
+        },
+        aliasAnalysisFromSchema()),
+};
+
+RegisterOperators regStrOps(createOperators(
+    stringOpGenArgs,
+    sizeof(stringOpGenArgs) / sizeof(stringOpGenArgs[0])));
```

Comment:
Are you doing the `sizeof()` calls to pass in the number of elements of the array to `createOperators()`?

Maybe make `stringOpGenArgs` a `std::array` instead, that way you can just call `.size()` instead.

Reply:
I was following the example here: https://fburl.com/tz3vkorc. I can try `std::array` instead.